controllers/helmchartinstall_controller.go

```diff
@@ -77,61 +126,255 @@ func (r *HelmChartInstallReconciler) render(ctx context.Context, claim *helmv1al
+		return nil, err
+	}
+
+	generatedMap.SetNamespace(claim.GetNamespace())
+
+	r.Log.V(0).Info("Generated config map to pass engine configuration", "configMap", generatedMap)
+
+	err = r.Client.Create(ctx, generatedMap)
+
+	if err != nil {
+		r.Log.V(0).Info("Error creating config map!", "claim", claim, "error", err, "configMap", generatedMap)
+		return nil, err
+	}
+
+	return generatedMap, err
+}
+
+// The main reason this exists as its own method is to encapsulate the hashing logic
+func (r *HelmChartInstallReconciler) generateConfigMap(name string, fileName string, fileContents string) (*corev1.ConfigMap, error) {
+	configMap := &corev1.ConfigMap{}
+	configMap.Name = name
+	configMap.Data = map[string]string{}
+
+	configMap.Data[fileName] = fileContents
+	h, err := hash.ConfigMapHash(configMap)
+	if err != nil {
+		r.Log.V(0).Info("Error hashing config map!", "error", err)
+		return configMap, err
+	}
+	configMap.Name = fmt.Sprintf("%s-%s", configMap.Name, h)
+
+	return configMap, nil
+}
+
+/**
+ * This is in its own method because it will involve
+ * reading through multiple levels of the configuration to see
+ * what the engine type is. It may even involve inferring an engine
+ * type based on the stack contents (though that may happen earlier
+ * in the lifecycle than this).
+ */
+func (r *HelmChartInstallReconciler) getEngineType(
+	claim *unstructured.Unstructured,
+	stackConfig *v1alpha1.StackConfiguration,
+) string {
+	return "helm2"
 }
 
-func (r *HelmChartInstallReconciler) processConfiguration(
+func (r *HelmChartInstallReconciler) executeBehavior(
 	ctx context.Context,
-	claim *helmv1alpha1.HelmChartInstall,
-	configuration unstructured.Unstructured,
-) (unstructured.Unstructured, error) {
-	// Given a claim and a stack configuration, transform the claim into something
-	// to inject into a stack's configuration renderer, and then render the Stack's
-	// resources from the configuration and the transformed claim
+	claim *unstructured.Unstructured,
+	engineConfiguration *corev1.ConfigMap,
+	targetStackImage string,
+	behavior *v1alpha1.StackConfigurationBehavior,
+) (*unstructured.Unstructured, error) {
+	for _, resource := range behavior.Resources {
+		// TODO error handling
+		// TODO use result
+		r.executeHook(ctx, claim, engineConfiguration, targetStackImage, resource)
+	}
+	// TODO return a real result
+	return &unstructured.Unstructured{}, nil
+}
+
+// TODO we could have a method create the job, and a higher-level one execute it.
+func (r *HelmChartInstallReconciler) executeHook(
+	ctx context.Context,
+	claim *unstructured.Unstructured,
+	engineConfig *corev1.ConfigMap,
+	targetStackImage string,
+	targetResourceDir string,
+) (*unstructured.Unstructured, error) {
+	// TODO if there is no config specified, either use an empty config or don't specify
+	// one at all.
+
 	ownerRef := meta.AsOwner(meta.ReferenceTo(claim, claim.GroupVersionKind()))
 	var jobBackoff int32
 	jobBackoff = 0
```

Comment:
I believe you could write `var jobBackoff int32 = 0` here, or just write `var jobBackoff int32` and leverage the fact that the zero value of an int32 is... zero.

Reply:
ooo, neat

Reply:
This is done